lib/srv/discovery/discovery.go

```diff
@@ -466,6 +472,75 @@ func (s *Server) Wait() error {
 	return nil
 }
 
+func (s *Server) watchCARotations(ctx context.Context) error {
+	clustername, err := s.Auth.GetClusterName()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	watcher, err := s.Auth.NewWatcher(ctx, types.Watch{
+		Kinds: []types.WatchKind{{
+			Kind: types.KindCertAuthority,
+			Filter: types.CertAuthorityFilter{
+				types.OpenSSHCA: clustername.GetName(),
+			}.IntoMap()}},
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	for {
+		select {
+		case event := <-watcher.Events():
+			if event.Type == types.OpInit {
```

Comment:
I'd still expect exactly one `OpInit` at the very beginning, outside of the loop, just as a matter of principle.